src/Summoner/Template/Stack.hs

```diff
@@ -12,7 +12,7 @@ import Summoner.Tree (TreeFs (..))
 
 
 stackFiles :: Settings -> [TreeFs]
-stackFiles Settings{..} = map createStackYaml settingsTestedVersions
+stackFiles Settings{..} = mapAndList createStackYaml settingsTestedVersions
```

Comment:
The implementation is now wrong here. You still need to add `memptyIfFalse settingsStack` because `stackFiles` function always works with `NonEmpty` list so the output list will be always non-empty as well. But it should be empty if users disabled `stack` build tool.